subprojects/packagefiles/luajit/src/meson.build

```
@@ -0,0 +1,342 @@
+ljver = meson.project_version().split('.')
+ljver = '.'.join(ljver[0], ljver[1])
+abiver = '5.1'
+verdir = 'luajit-' + ljver
+jitdir = get_option('jitdir')
+if jitdir == ''
+  jitdir = get_option('datadir') / verdir / 'jit'
+endif
+
+foreach _var, _subdir : {
+  'install_cmod': 'libdir',
+  'install_lmod': 'datadir',
+}
+  _val = get_option(_var)
+  if _val != ''
+    assert(not fs.is_absolute(_val))
+  else
+    _val = get_option(_subdir) / 'lua' / abiver
+  endif
+  set_variable(_var, _val)
+endforeach
+
+luajit_h = custom_target(
+  'luajit.h',
+  capture: true,
+  command: [
+    'sed',
+    '-e', 's/@0@/@1@/'.format(
+      'ROLLING',
+      meson.project_version().split('.')[-1],
+    ),
+    '-e', '/^#error /d',
+    '@INPUT@',
+  ],
+  input: 'luajit_rolling.h',
+  output: 'luajit.h',
+  install: true,
+  install_tag: 'devel',
+  install_dir: get_option('includedir') / verdir,
+)
+
+subdir('host')
+
+if host_machine.system() == 'windows'
+  lj_vm_out = custom_target(
+    'lj_vm.o',
+    output: 'lj_vm.o',
+    command: [
+      buildvm_exe,
+      '-m', LJVM_MODE,
+      '-o', '@OUTPUT@',
+    ],
+  )
+else
+  lj_vm_out = custom_target(
+    'lj_vm.S',
+    output: 'lj_vm.S',
+    command: [
+      buildvm_exe,
+      '-m', LJVM_MODE,
+      '-o', '@OUTPUT@',
+    ],
+  )
+endif
```

The only different is file extension. To make it more readable, you could just set the name up front.

```suggestion
lj_vm = host_machine.system() == 'windows' ? 'lj_vm.o' : 'lj_vm.S'
lj_vm_out = custom_target(
  lj_vm ,
  output: lj_vm ,
  command: [
    buildvm_exe,
    '-m', LJVM_MODE,
    '-o', '@OUTPUT@',
  ],
)
```

---

Thanks, amended.